src/main/java/frc/robot/subsystems/ground_manager/intake/Intake.java

```
@@ -0,0 +1,168 @@
+
+  private final TalonFX intakeMotor;
+  private final TalonFXConfiguration motor_config = new TalonFXConfiguration()
+      .withSlot0(new Slot0Configs().withKP(IntakeConstants.P).withKI(IntakeConstants.I).withKD(IntakeConstants.D)
+          .withKG(IntakeConstants.G).withGravityType(GravityTypeValue.Arm_Cosine))
+      .withFeedback(new FeedbackConfigs().withSensorToMechanismRatio((8.0357 / 1.0)));
+  private double intakePosition;
+  private final double tolerance;
+
+  private MotionMagicVoltage motor_request = new MotionMagicVoltage(0).withSlot(0);
+
+
+  private Intake() {
+    super(IntakeStates.IDLE);
+    motor_config.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+    motor_config.MotorOutput.Inverted = InvertedValue.CounterClockwise_Positive;
+    intakeMotor = new TalonFX(Ports.IntakePorts.INTAKE_MOTOR);
+    intakeMotor.getConfigurator().apply(motor_config);
+    motor_config.MotionMagic.MotionMagicCruiseVelocity = IntakeConstants.MotionMagicCruiseVelocity;
+    motor_config.MotionMagic.MotionMagicAcceleration = IntakeConstants.MotionMagicAcceleration;
+    motor_config.MotionMagic.MotionMagicJerk = IntakeConstants.MotionMagicJerk;
+    tolerance = 0.0;
+  }
+  public double getIntakePosition() {
+    return intakePosition;
+}
+
+  public boolean atGoal() {
+    return switch (getState()) {
+          case IDLE ->
+            MathUtil.isNear(IntakePositions.IDLE, intakePosition, tolerance);
+          case INTAKING ->
+            MathUtil.isNear(IntakePositions.INTAKING, intakePosition, tolerance);
+          case CLIMB ->
+            MathUtil.isNear(IntakePositions.CLIMB, intakePosition, tolerance);
+          case HANDOFF ->
+            MathUtil.isNear(IntakePositions.HANDOFF, intakePosition, tolerance);
+          case SCORE_L1 ->
+            MathUtil.isNear(IntakePositions.SCORE_L1, intakePosition, tolerance);
+          default -> throw new IllegalArgumentException("Unexpected value: " + getState());
```

I don't know if we need to throw an error here... we could just return a default value instead. Don't want to end the robot code for calling `atGoal()` when not in one of the above states.